lib/plugins/filter/before_post_render/indented_code_block.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const { escapeHTML, stripIndent } = require('hexo-util');
+
+const rIndentedCodeBlock = /^(?:((?:(?: {0,3}>){1,3}[^\S\r\n])?)( {4}|\t)([^\n]*?)(\n|$))+/gm;
+
+function indentedCodeBlock(data) {
+  data.content = data.content.replace(rIndentedCodeBlock, (content, start, $2, $3, end) => {
+
+    // PR #3765
+    if (start.includes('>')) {
+      const depth = start.split('>').length - 1;
+      const regexp = new RegExp(`^( {0,3}>){0,${depth}}([^\\S\\r\\n]|$)`, 'mg');
+      const paddingOnEnd = ' '; // complement uncaptured whitespaces at last line
+      content = (content + paddingOnEnd).replace(regexp, '').replace(/\n$/, '');
+    }
+
+    content = escapeHTML(stripIndent(content))
+      .replace(/{/g, '&#123;')
+      .replace(/}/g, '&#125;');
+
+    content = `<pre><code>${content}</code></pre>`;
+
+    return `${start}<!--hexoPostRenderEscape:${content}:hexoPostRenderEscape-->${end}`;
```

Comment:
```suggestion
return start
  + '<!--hexoPostRenderEscape:<pre><code>'
  + escapeHTML(stripIndent(content))
    .replace(/{/g, '&#123;')
    .replace(/}/g, '&#125;')
  + '</code></pre>:hexoPostRenderEscape-->'
  + end;
```

Reply:
Avoid unnecessary `content` reassignment without breaking code readability.

Reply:
I agree your advice about avoiding reassignment.
But I think it is also important that the code shows the process step-by-step.
So I choice to use new variables instead of reassignment.